examples/rl/DDPG_Pendulum.py

```diff
@@ -0,0 +1,428 @@
+Gradient Zero Because its "tanh"!!
+"""
+
+
+def get_actor():
+
+    init = tf.keras.initializers.GlorotNormal(seed=1)
+    last_init = tf.random_uniform_initializer(minval=-0.003, maxval=0.003)
+
+    inputs = layers.Input(shape=(nS))
+    out = layers.Dense(512, activation="relu", kernel_initializer=init)(inputs)
+    out = layers.LayerNormalization()(out)
+    out = layers.Dense(512, activation="relu", kernel_initializer=init)(out)
+    out = layers.LayerNormalization()(out)
+    outputs = layers.Dense(1, activation="tanh", kernel_initializer=last_init)(out)
+
+    # our upper bound is 2.0 for Pendulum.
+    # This scale out our Actions
+    outputs = outputs * upper_bound
+    model = tf.keras.Model(inputs, outputs)
+    return model
+
+
+def get_critic():
+
+    init = tf.keras.initializers.GlorotNormal(seed=1)
+
+    # state as input
+    input_s = layers.Input(shape=(nS))
+    state_h1 = layers.Dense(24, activation="relu", kernel_initializer=init)(input_s)
+    state_h1 = layers.LayerNormalization()(state_h1)
+    state_h2 = layers.Dense(48, activation="relu", kernel_initializer=init)(state_h1)
+    state_h2 = layers.LayerNormalization()(state_h2)
+
+    # Action as input
+    input_a = layers.Input(shape=(1))
+    action_h1 = layers.Dense(48, activation="relu", kernel_initializer=init)(input_a)
+    action_h1 = layers.LayerNormalization()(action_h1)
+
+    # Both are passed through seperate layer before concatenating.
+    merged = layers.Concatenate()([input_s, input_a])
+
+    out = layers.Dense(512, activation="relu", kernel_initializer=init)(merged)
+    out = layers.LayerNormalization()(out)
+    out = layers.Dense(512, activation="relu", kernel_initializer=init)(out)
+    out = layers.LayerNormalization()(out)
+    outputs = layers.Dense(1, kernel_initializer=init)(out)
+
+    # Outputs single value for give State-Action
+    model = tf.keras.Model([input_s, input_a], outputs)
+
+    return model
+
+
+"""
+Policy give Action given by out Actor Network plus some Exploratory Noise for
+exploration.
+"""
+
+
+def policy(state, noise_obj):
+
+    sampled_nums = tf.squeeze(actor_model(state))
+    noise = noise_obj()
+    sampled_nums = sampled_nums.numpy() + noise
+
+    # We make sure action is within bounds
+    legal_action = np.clip(sampled_nums, lower_bound, upper_bound)
+
+    return [np.squeeze(legal_action)]
+
+
+"""
+**HYPER PARAMETERS and OBJECT Declaration**
+"""
+
+stddev = 0.2
+ou_noise = OUActionNoise(mu=np.zeros(1), sigma=float(stddev) * np.ones(1))
+
+actor_model = get_actor()
+critic_model = get_critic()
+
+target_actor = get_actor()
+target_critic = get_critic()
+
+# Making the Weights same at start
+target_actor.set_weights(actor_model.get_weights())
+target_critic.set_weights(critic_model.get_weights())
+
+c_lr = 0.002
+a_lr = 0.001
+
+critic_optimizer = tf.keras.optimizers.Adam(c_lr)
+actor_optimizer = tf.keras.optimizers.Adam(a_lr)
+
+total_eps = 100
+GAMMA = 0.99
```

Comment:
Why is this variable in all caps and the other lowercase? Use all caps for global constants defined at the start of the file only.